custom_enumerable_methods.rb

```diff
@@ -0,0 +1,261 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    x = 0
+    self_class = self.class
+    array = if self_class == Array
+              self
+            elsif self_class == Range
+              to_a
+            else
+              flatten
+            end
+    while x < array.length
+      if self_class == Hash
+        yield array[x], array[x + 1]
+        x += 2
+      else
+        yield array[x]
+        x += 1
+      end
+    end
+  end
```

Comment:
`my_each` identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods)) ``
- After calling the given block once for each element in self, returns the array itself
```ruby
[1, 2, 3].my_each {|num| num}  #should return [1, 2, 3]
```